src/map/basic_map.rs

```diff
@@ -110,6 +117,43 @@ impl Map {
     self.stairs_up = Some(Point::new(stairs_position.0, stairs_position.1));
   }
 
+  pub fn add_doors_to_rooms(&mut self) {
+    for i in 0..self.rooms.len() {
+      for x in (self.rooms[i as usize].x1)..(self.rooms[i as usize].x2 + 2) {
```

Comment:
to loop over all numbers in an inclusive range you can use `..=` instead of `..`

Reply:
also, why `+ 2`?

Reply:
I may be incorrect, but from testing it seemed like the x1/y1 positions for the rooms when indexed are wall tiles, but the x2/y2 positions are not?  When I remove the offsets it will place the doors correctly (within the wall bounds) on the left and upper walls, but place them incorrectly (inside of the room) on the right and bottom walls.

That's specifically for the use of the +1.  The +2 could be removed by using `..=` as you suggested.

Reply:
interesting.. that speaks to a deeper bug in the map generation I think.